unit_tests/tests/trigger/test_trigger_decoder.cpp

```diff
@@ -1192,3 +1192,69 @@ TEST(big, testAssertWeAreNotMissingASpark299) {
 
 	ASSERT_EQ( 0,  unitTestWarningCodeState.recentWarnings.getCount()) << "warningCounter#1";
 }
+
+TEST(big, testSparkDwell) {
+	printf("*************************************************** testSparkDwell\r\n");
+
+	EngineTestHelper eth(engine_type_e::TEST_ENGINE);
+	engineConfiguration->ignitionMode = IM_WASTED_SPARK;
+	setupSimpleTestEngineWithMafAndTT_ONE_trigger(&eth);
+	engineConfiguration->isIgnitionEnabled = true;
+	engineConfiguration->isInjectionEnabled = false;
+
+	ASSERT_EQ( 0,  unitTestWarningCodeState.recentWarnings.getCount()) << "warningCounter#0";
+
+	eth.smartFireRise(20);
+	ASSERT_EQ( 0,  engine->triggerCentral.triggerState.currentCycle.current_index) << "ci#0";
+
+	eth.smartFireFall(20);
+	ASSERT_EQ( 1,  engine->triggerCentral.triggerState.currentCycle.current_index) << "ci#1";
+
+	eth.smartFireRise(20);
+	ASSERT_EQ( 0,  engine->triggerCentral.triggerState.currentCycle.current_index) << "ci#2";
+
+	eth.smartFireFall(20);
+	ASSERT_EQ( 1,  engine->triggerCentral.triggerState.currentCycle.current_index) << "ci#3";
+
+	eth.smartFireRise(20);
+
+	eth.smartFireFall(20);
+	ASSERT_EQ( 1,  eth.engine.triggerCentral.triggerState.currentCycle.current_index) << "ci#5";
+
+	ASSERT_EQ(3000, Sensor::getOrZero(SensorType::Rpm));
```

Comment:
Your call if you want to use eth.assertRpm(3000); just know that it exists

Reply:
I would suggest to get rid of such wrappers at all :)
See #6477